confmap/resolver.go

```diff
@@ -115,28 +115,21 @@ func (mr *Resolver) Resolve(ctx context.Context) (*Conf, error) {
 		// For backwards compatibility:
 		// - empty url scheme means "file".
 		// - "^[A-z]:" also means "file"
-		scheme := "file"
-		if idx := strings.Index(uri, ":"); idx != -1 && !driverLetterRegexp.MatchString(uri) {
-			scheme = uri[:idx]
-		} else {
-			uri = scheme + ":" + uri
+		if driverLetterRegexp.MatchString(uri) {
+			uri = "file:" + uri
 		}
-		p, ok := mr.providers[scheme]
-		if !ok {
-			return nil, fmt.Errorf("scheme %q is not supported for uri %q", scheme, uri)
-		}
-		ret, err := p.Retrieve(ctx, uri, mr.onChange)
+		ret, err := mr.retrieveValue(ctx, uri, "file")
```

Comment:
Nit: it is a bit unclear what "file" means here. Also, two string parameters are easy to support in the wrong order. s an alternate pass a struct with 2 fields, which will also make clear what the meaning of the parameters is because fields have names.

Reply:
Commented on the param. Let me know if you think the struct will be very useful and I can do it.

Reply:
Done.